### PR TITLE
add train_mode to encode

### PR DESCRIPTION
**Pull request by fulaphex**
I'm using the `SentenceTransformer` as a layer in a model I'm working on. I want to finetune the transformer layer in the training, so I need this modification to the `encode` function so that gradients are being computed.

### CODE REVIEW

**Comment by nreimers** [comment]
Hi @fulaphex 
Thanks for proposing this PR.

Not sure if a 'train_mode' flag in the encode function would be the best way. 

The SentenceTransformer object inherits from nn.Sequential, i.e., it has a regular forward function which can by used within pytorch.

You can get a trainable sentence embedding like this:
```python
from sentence_transformers import SentenceTransformer, util
model = SentenceTransformer('stsb-roberta-base')


batch = ["Hello World", "This is a test"]

#Get the features for the batch, i.e. tokenize them
features = model.tokenize(batch)

#Move features to the device
features = util.batch_to_device(features, model.device)

#Call the forward function of the model
output = model(features)

#The model returns various outputs, where 'sentence_embedding' is your sentence embedding
embeddings = output['sentence_embedding']
print(embeddings)
```
